### ISSUE TITLE
[release] First early-release fails for tag deletion

### ISSUE THREAD

**Issue by manikmagar** [jreleaser/jreleaser#1739]
We upgraded to use 1.14.0, and we ran into this error.

This is a new[ test repo project](https://github.com/avioconsulting/jreleaser-test-app/blob/main/jreleaser.yml); its first early release, but it is failing to delete a non-existent early-access, tag. Am I missing any configuration that may have changed in a new release?

```
[INFO]    [upload] Uploading is not enabled. Skipping
[INFO]  Releasing to https://github.com/avioconsulting/jreleaser-test-app@main
[DEBUG] fetching all branches on avioconsulting/jreleaser-test-app
[DEBUG] looking up release with tag early-access at repository avioconsulting/jreleaser-test-app
[DEBUG] fetching release on avioconsulting/jreleaser-test-app with tag early-access
[DEBUG] release early-access does not exist
[DEBUG] creating release early-access
[DEBUG] deleting tag early-access from avioconsulting/jreleaser-test-app
jreleaser.shadow.org.jreleaser.sdk.commons.RestAPIException: 404: Not Found
{"message":"Not Found","documentation_url":"https://docs.github.com/rest/git/refs#delete-a-reference","status":"404"}
	at jreleaser.shadow.org.jreleaser.sdk.commons.ClientUtils.lambda$builder$1(ClientUtils.java:120)
	at jreleaser.shadow.feign.InvocationContext.decodeError(InvocationContext.java:126)

```

_Originally posted by @manikmagar in https://github.com/jreleaser/jreleaser/discussions/1738_

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. A new Java project to publish to Maven Central and Release to Github
2. Full release with JReleaser 1.14.0

### Expected Behaviour

The 1.0.0-SNAPSHOT release should create an early-access release and tag.

### Actual Behaviour

Tag creation fails when checking for existing tag. Since this is a first time, tag doesn't exist yet.

### Environment Information

[INFO]  JReleaser 1.14.0
------------------------------------------------------------
jreleaser 1.14.0
------------------------------------------------------------
Build time:   2024-08-31 06:25:38Z
Revision:     4a8bb20f77d2aced0e1865698c484c2a6cf25e7b
JVM:          17.0.12 (Azul Systems, Inc. 17.0.12+7-LTS)
------------------------------------------------------------

Please paste stacktraces from `[out|build|target]/jreleaser/trace.log`.

Actions Run - https://github.com/avioconsulting/jreleaser-test-app/actions/runs/11177008155

```
[DEBUG]     [nexus2] Expect: [100-continue]
[DEBUG]     [nexus2] Accept: [*/*]
[DEBUG]     [nexus2] User-Agent: [JReleaser/1.14.0]
[DEBUG]     [nexus2] Content-Type: [text/plain]
[DEBUG]     [nexus2] sending data
[DEBUG]     [nexus2] handling response
[INFO]  Uploading distributions and files
[INFO]    [upload] Uploading is not enabled. Skipping
[INFO]  Releasing to https://github.com/avioconsulting/jreleaser-test-app@main
[DEBUG] fetching all branches on avioconsulting/jreleaser-test-app
[DEBUG] looking up release with tag early-access at repository avioconsulting/jreleaser-test-app
[DEBUG] fetching release on avioconsulting/jreleaser-test-app with tag early-access
[DEBUG] release early-access does not exist
[DEBUG] creating release early-access
[DEBUG] deleting tag early-access from avioconsulting/jreleaser-test-app
jreleaser.shadow.org.jreleaser.sdk.commons.RestAPIException: 404: Not Found
{"message":"Not Found","documentation_url":"https://docs.github.com/rest/git/refs#delete-a-reference","status":"404"}
	at jreleaser.shadow.org.jreleaser.sdk.commons.ClientUtils.lambda$builder$1(ClientUtils.java:120)
	at jreleaser.shadow.feign.InvocationContext.decodeError(InvocationContext.java:126)
	at jreleaser.shadow.feign.InvocationContext.proceed(InvocationContext.java:72)
	at jreleaser.shadow.feign.RedirectionInterceptor.intercept(RedirectionInterceptor.java:44)
	at jreleaser.shadow.feign.ResponseInterceptor.lambda$apply$2(ResponseInterceptor.java:69)
	at jreleaser.shadow.feign.ResponseHandler.handleResponse(ResponseHandler.java:63)
	at jreleaser.shadow.feign.SynchronousMethodHandler.executeAndDecode(SynchronousMethodHandler.java:114)
	at jreleaser.shadow.feign.SynchronousMethodHandler.invoke(SynchronousMethodHandler.java:70)
	at jreleaser.shadow.feign.ReflectiveFeign$FeignInvocationHandler.invoke(ReflectiveFeign.java:99)
	at jdk.proxy2/jdk.proxy2.$Proxy20.deleteTag(Unknown Source)
	at jreleaser.shadow.org.jreleaser.sdk.github.Github.deleteTag(Github.java:328)
	at jreleaser.shadow.org.jreleaser.sdk.github.GithubReleaser.deleteTags(GithubReleaser.java:535)
	at jreleaser.shadow.org.jreleaser.sdk.github.GithubReleaser.createRelease(GithubReleaser.java:338)
	at jreleaser.shadow.org.jreleaser.sdk.github.GithubReleaser.createRelease(GithubReleaser.java:232)
	at jreleaser.shadow.org.jreleaser.sdk.git.release.AbstractReleaser.release(AbstractReleaser.java:59)
	at jreleaser.shadow.org.jreleaser.engine.release.Releasers.release(Releasers.java:56)
	at jreleaser.shadow.org.jreleaser.workflow.ReleaseWorkflowItem.doInvoke(ReleaseWorkflowItem.java:39)
	at jreleaser.shadow.org.jreleaser.workflow.AbstractWorkflowItem.lambda$invoke$0(AbstractWorkflowItem.java:43)
	at jreleaser.shadow.org.jreleaser.engine.hooks.HookExecutor.execute(HookExecutor.java:70)
	at jreleaser.shadow.org.jreleaser.workflow.AbstractWorkflowItem.invoke(AbstractWorkflowItem.java:43)
	at jreleaser.shadow.org.jreleaser.workflow.WorkflowImpl.doExecute(WorkflowImpl.java:129)
	at jreleaser.shadow.org.jreleaser.workflow.WorkflowImpl.execute(WorkflowImpl.java:54)
	at jreleaser.shadow.org.jreleaser.cli.FullRelease.doExecute(FullRelease.java:220)
	at jreleaser.shadow.org.jreleaser.cli.AbstractModelCommand.execute(AbstractModelCommand.java:90)
	at jreleaser.shadow.org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:52)
	at jreleaser.shadow.org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:37)
	at jreleaser.shadow.picocli.CommandLine.executeUserObject(CommandLine.java:2045)
	at jreleaser.shadow.picocli.CommandLine.access$1500(CommandLine.java:148)
	at jreleaser.shadow.picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2465)
	at jreleaser.shadow.picocli.CommandLine$RunLast.handle(CommandLine.java:2457)
	at jreleaser.shadow.picocli.CommandLine$RunLast.handle(CommandLine.java:2419)
	at jreleaser.shadow.picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at jreleaser.shadow.picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
	at jreleaser.shadow.picocli.CommandLine.execute(CommandLine.java:2174)
	at jreleaser.shadow.org.jreleaser.cli.Main.execute(Main.java:98)
	at jreleaser.shadow.org.jreleaser.cli.Main.run(Main.java:94)
	at org.jreleaser.tool.JReleaser.main(JReleaser.java:47)
[INFO]  Writing output properties to out/jreleaser/output.properties
[ERROR] JReleaser failed after 46.953 s
org.jreleaser.model.JReleaserException: Unexpected error when creating release
	at jreleaser.shadow.org.jreleaser.workflow.ReleaseWorkflowItem.doInvoke(ReleaseWorkflowItem.java:41)
	at jreleaser.shadow.org.jreleaser.workflow.AbstractWorkflowItem.lambda$invoke$0(AbstractWorkflowItem.java:43)
	at jreleaser.shadow.org.jreleaser.engine.hooks.HookExecutor.execute(HookExecutor.java:70)
	at jreleaser.shadow.org.jreleaser.workflow.AbstractWorkflowItem.invoke(AbstractWorkflowItem.java:43)
	at jreleaser.shadow.org.jreleaser.workflow.WorkflowImpl.doExecute(WorkflowImpl.java:129)
	at jreleaser.shadow.org.jreleaser.workflow.WorkflowImpl.execute(WorkflowImpl.java:54)
	at jreleaser.shadow.org.jreleaser.cli.FullRelease.doExecute(FullRelease.java:220)
	at jreleaser.shadow.org.jreleaser.cli.AbstractModelCommand.execute(AbstractModelCommand.java:90)
	at jreleaser.shadow.org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:52)
	at jreleaser.shadow.org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:37)
	at jreleaser.shadow.picocli.CommandLine.executeUserObject(CommandLine.java:2045)
	at jreleaser.shadow.picocli.CommandLine.access$1500(CommandLine.java:148)
	at jreleaser.shadow.picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2465)
	at jreleaser.shadow.picocli.CommandLine$RunLast.handle(CommandLine.java:2457)
	at jreleaser.shadow.picocli.CommandLine$RunLast.handle(CommandLine.java:2419)
	at jreleaser.shadow.picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at jreleaser.shadow.picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
	at jreleaser.shadow.picocli.CommandLine.execute(CommandLine.java:2174)
	at jreleaser.shadow.org.jreleaser.cli.Main.execute(Main.java:98)
	at jreleaser.shadow.org.jreleaser.cli.Main.run(Main.java:94)
	at org.jreleaser.tool.JReleaser.main(JReleaser.java:47)
Caused by: jreleaser.shadow.org.jreleaser.model.spi.release.ReleaseException: jreleaser.shadow.org.jreleaser.sdk.commons.RestAPIException: 404: Not Found
{"message":"Not Found","documentation_url":"https://docs.github.com/rest/git/refs#delete-a-reference","status":"404"}
	at jreleaser.shadow.org.jreleaser.sdk.github.GithubReleaser.createRelease(GithubReleaser.java:236)
	at jreleaser.shadow.org.jreleaser.sdk.git.release.AbstractReleaser.release(AbstractReleaser.java:59)
	at jreleaser.shadow.org.jreleaser.engine.release.Releasers.release(Releasers.java:56)
	at jreleaser.shadow.org.jreleaser.workflow.ReleaseWorkflowItem.doInvoke(ReleaseWorkflowItem.java:39)
	... 20 more
Caused by: jreleaser.shadow.org.jreleaser.sdk.commons.RestAPIException: 404: Not Found
{"message":"Not Found","documentation_url":"https://docs.github.com/rest/git/refs#delete-a-reference","status":"404"}
	at jreleaser.shadow.org.jreleaser.sdk.commons.ClientUtils.lambda$builder$1(ClientUtils.java:120)
	at jreleaser.shadow.feign.InvocationContext.decodeError(InvocationContext.java:126)
	at jreleaser.shadow.feign.InvocationContext.proceed(InvocationContext.java:72)
	at jreleaser.shadow.feign.RedirectionInterceptor.intercept(RedirectionInterceptor.java:44)
	at jreleaser.shadow.feign.ResponseInterceptor.lambda$apply$2(ResponseInterceptor.java:69)
	at jreleaser.shadow.feign.ResponseHandler.handleResponse(ResponseHandler.java:63)
	at jreleaser.shadow.feign.SynchronousMethodHandler.executeAndDecode(SynchronousMethodHandler.java:114)
	at jreleaser.shadow.feign.SynchronousMethodHandler.invoke(SynchronousMethodHandler.java:70)
	at jreleaser.shadow.feign.ReflectiveFeign$FeignInvocationHandler.invoke(ReflectiveFeign.java:99)
	at jdk.proxy2/jdk.proxy2.$Proxy20.deleteTag(Unknown Source)
	at jreleaser.shadow.org.jreleaser.sdk.github.Github.deleteTag(Github.java:328)
	at jreleaser.shadow.org.jreleaser.sdk.github.GithubReleaser.deleteTags(GithubReleaser.java:535)
	at jreleaser.shadow.org.jreleaser.sdk.github.GithubReleaser.createRelease(GithubReleaser.java:338)
	at jreleaser.shadow.org.jreleaser.sdk.github.GithubReleaser.createRelease(GithubReleaser.java:232)
	... 23 more

```

**Comment by manikmagar** [comment]
I have tested as far back as 1.6.0 for this, and all are failing with the same error.

**Comment by aalmiray** [comment]
Code at `org.jreleaser.sdk.github.Github.deleteTag(Github.java:328)` should handle `404` without failure. 

Deleting a non-existent `early-access` tag (by checking if the project is snapshot) should not be cause for an error. Rather, the error may be logged in debug with a message such as "tag <tag> does not exist" and continue.

**Comment by manikmagar** [comment]
Not failing for 404 makes sense. 

The puzzling part is why failing now? Wasn't most if not everyone who onboarded to Jreleaser so far were missing the early-release tag initially?  Even we onboarded many repos and never ran into this before.

**Comment by aalmiray** [comment]
Exactly. Which makes me think a rule was updated in GitHub's backend.

**Comment by manikmagar** [comment]
I ran following commands using `gh` cli - 

Deleting an existing tag, returns success.
```
gh api --method DELETE -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" /repos/{owner}/{repo}/git/refs/tags/temp-v2
```

Deleting non-existing tag, returns 422
```
gh api --method DELETE -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" /repos/{owner}/{repo}/git/refs/tags/temp-v3

gh: Reference does not exist (HTTP 422)
{
  "message": "Reference does not exist",
  "documentation_url": "https://docs.github.com/rest/git/refs#delete-a-reference",
  "status": "422"
}
```

HTTP 404 is only returned if the target URL is incorrect. Is it possible for jreleaser logging to print verbose logs of git requests being made?

**Comment by aalmiray** [comment]
You may use `jreleaser.feign.logger.level` as described at https://jreleaser.org/guide/latest/reference/environment.html#_magic_properties

**Comment by manikmagar** [comment]
This is blocking us from using JReleaser on new repos. I have just sent a PR to fix this issue.

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.15.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.15.0))
